### DISCUSSION TITLE
Possible to support 2 spaces for nested list indentation?

### DISCUSSION THREAD

**Discussion by khughitt** [markedjs/marked#3015] [General]
Greetings!

I was wondering if is it possible to support using 2 spaces for nested list levels with marked + gfm?

From the [gfm spec](https://github.github.com/gfm/#example-272):

>  A sublist must be indented the same number of spaces a paragraph would need to be in order to be included in the list item.

They give an example where nesting uses 2 spaces.

When using marked to render markdown, I find that it works as expected when either tabs or 4 spaces are used for each level of indentation, but if only 2 spaces are used, the nesting breaks.

Example of expected behavior:

List with 4 spaces:

- one
    - two

List with 2 spaces:

- one
  - two

List with tabs:

- one
	- two

**Comment by UziTech** [comment]
It seems to work fine in our [demo](https://marked.js.org/demo/?text=List%20with%204%20spaces%3A%0A%0A-%20one%0A%20%20%20%20-%20two%0A%0AList%20with%202%20spaces%3A%0A%0A-%20one%0A%20%20-%20two%0A%0AList%20with%20tabs%3A%0A%0A-%20one%0A%09-%20two&options=%7B%0A%20%22async%22%3A%20false%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22hooks%22%3A%20null%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%0A%7D&version=9.0.3). Can you create a demo that doesn't work?

**Reply by khughitt** [reply]
Hey!

Apologies - I somehow completely missed your response.

I think the issue may actually be limited to mixed ordered/unordered lists, e.g. [here](https://marked.js.org/demo/?text=Mixed%20list%20(4%20spaces)%3A%0A%0A1.%20one%0A%20%20%20%20-%20a%0A%20%20%20%20-%20b%0A2.%20two%0A%0AMixed%20list%20(2%20spaces)%3A%0A%0A1.%20one%0A%20%20-%20a%0A%20%20-%20b%0A2.%20two&options=%7B%0A%20%22headerIds%22%3A%20false%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22mangle%22%3A%20false%0A%7D&version=9.0.3).

I'll follow up if I find/can reproduce the issue for lists of a single type.

Side note: nice work with the URL-hashed demos! That makes it very easy to show things like this.

**Reply by UziTech** [reply]
It looks like marked renders the same as CommonMark. If this is something you think should be fixed you will need to fix it in CommonMark first as we strive to match their output.

[CommonMark demo](https://spec.commonmark.org/dingus/?text=Mixed%20list%20(4%20spaces)%3A%0A%0A1.%20one%0A%20%20%20%20-%20a%0A%20%20%20%20-%20b%0A2.%20two%0A%0AMixed%20list%20(2%20spaces)%3A%0A%0A1.%20one%0A%20%20-%20a%0A%20%20-%20b%0A2.%20two)

**Reply by khughitt** [reply]
Ah okay. I checked the [spec](https://spec.commonmark.org/0.30) and it looks like the indentation required on a subsequent line incorporates the length of the list marker..

Since the list marker goes from 1 character ("-") to two characters ("n.") in my mixed/nested example, the required indentation goes from 2 -> 3 characters.

So this is indeed expected behavior.

Thanks again for taking the time to follow up! It's appreciated.
